hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationThread.java

```diff
@@ -104,6 +104,7 @@ public final void run() {
                 try {
                     task = queue.take(priority);
                 } catch (InterruptedException e) {
+                    currentThread().interrupt();
```

Comment:
In this case restoring the interrupt isn't strictly needed. This logic is in the main loop of the OperationThread. And this main loop will check the 'shutdown' flag which is set before the interrupt flag is set. 

```
  public final void shutdown() {
        shutdown = true;
        interrupt();
    }
```

In other words; even if the interrupt flag is consumed, the Thread will not process any further tasks and terminate.